frustratometer/optimization/optimization.py

```diff
@@ -58,19 +58,20 @@ def compute_swap_energy(seq_index, model_h, model_J, mask, pos1, pos2):
 
 
 @numba.njit
-def sequence_mutation(seq_index, model_h, model_J, mask):
+def sequence_mutation(seq_index, model_h, model_J, mask,valid_indices=[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20]):
     seq_index = seq_index.copy()
-    r = np.random.randint(0, len(seq_index)*20)  # Select a random index    
-    res = r // 20
-    aa_new = r % 20 + 1
+    r = np.random.randint(0, len(valid_indices)*len(seq_index)) # Select a random index
+    res = r // len(valid_indices)
+    aa_new_index = valid_indices[r % len(valid_indices)]
+    aa_new = _AA[aa_new_index] 
```

Comment:
Are you using `aa_new` here? if not rename `aa_new_index` to `aa_new` and remove the current `aa_new`